lib/taskana-core/src/main/java/pro/taskana/task/internal/TaskCommentServiceImpl.java

```diff
@@ -136,7 +142,10 @@ void deleteTaskComment(String taskCommentId)
           || taskanaEngine.getEngine().isUserInRole(TaskanaRole.ADMIN)
           || taskanaEngine.getEngine().isUserInRole(TaskanaRole.TASK_ADMIN)) {
 
+        Task task =
+            taskService.getTask(taskCommentMapper.findById(taskCommentId).getTaskId());
         taskCommentMapper.delete(taskCommentId);
+        taskMapper.decrementNumberOfComments(task.getId(), Instant.now());
```

Comment:
if taskCommentMapper.delete(taskCommentId) fails we do get an exception and this code is not executed, right?

Reply:
yes, from how i understand it, taskCommentMapper.delete(taskCommentId) throws an exception if the delete operation fails. If that method call throws an exceptinon, taskMapper.decrementNumberOfComments(task.getId(), Instant.now()) is not executed.